examples/aerial_atomic_swap.py

```diff
@@ -0,0 +1,241 @@
+
+        # Create Bob's token
+        mint_single_msg = {
+            "mint_single": {
+                "to_address": str(bob.address()),
+                "id": TOKEN_ID_2,
+                "supply": "2000",
+                "data": "some_data",
+            },
+        }
+
+        print(f"Minting tokens (id: {TOKEN_ID_1})...")
+        tx = contract.execute(mint_single_msg, alice).wait_to_complete()
+        print(f"Minted tokens (id: {TOKEN_ID_1} tx: {tx.tx_hash})")
+
+    # Step 2. Query what the current balance state is
+    result = contract.query(
+        {
+            "balance_batch": {
+                "addresses": [
+                    {
+                        "address": str(bob.address()),
+                        "id": TOKEN_ID_1,
+                    },
+                    {
+                        "address": str(alice.address()),
+                        "id": TOKEN_ID_2,
+                    },
+                ]
+            }
+        }
+    )
+    print(f'Alice has {result["balances"][1]} of Bob\'s tokens')
+    print(f'Bob has {result["balances"][0]} of Alice\'s tokens')
+
+    # Step 3. Perform the atomic swap using the transaction builder and the client
+
+    # Perform an atomic swap. Since this is a specialised operation we need to be able to manually
+    # build up the contents of the transaction
+    tx = Transaction()
+
+    # add the first message which is instructing the transfer of Alice's token to Bob
+    tx.add_message(
+        create_cosmwasm_execute_msg(
+            alice.address(),
+            contract.address,
+            {
+                "transfer_single": {
+                    "operator": str(alice.address()),
+                    "from_address": str(alice.address()),
+                    "to_address": str(bob.address()),
+                    "id": TOKEN_ID_1,
+                    "value": "1",
+                },
+            },
+        )
+    )
+
+    # add the second message which is instructing the transfer of Bob's token to Alice
+    tx.add_message(
+        create_cosmwasm_execute_msg(
+            bob.address(),
+            contract.address,
+            {
+                "transfer_single": {
+                    "operator": str(bob.address()),
+                    "from_address": str(bob.address()),
+                    "to_address": str(alice.address()),
+                    "id": TOKEN_ID_2,
+                    "value": "1",
+                },
+            },
+        )
+    )
+
+    # lookup the account information for each for Alice and Bob
+    alice_account = client.query_account(alice.address())
+    bob_account = client.query_account(bob.address())
+
+    # gas and fee estimation
+    gas_limit = 300000
+    fee = client.estimate_fee_from_gas(gas_limit)
+
+    # seal the transaction: this stops all further updates to the transactions and this is the step
+    # where additional metadata is added like gas, fees and the individual signers sequence numbers
+    tx.seal(
```

Comment:
Maybe we should make sure that we can sign only sealed TXs.

Reply:
Fixed